numpy/core/_methods.py

```diff
@@ -56,16 +59,23 @@ def _any(a, axis=None, dtype=None, out=None, keepdims=False):
 def _all(a, axis=None, dtype=None, out=None, keepdims=False):
     return umr_all(a, axis, dtype, out, keepdims)
 
-def _count_reduce_items(arr, axis):
-    if axis is None:
-        axis = tuple(range(arr.ndim))
-    if not isinstance(axis, tuple):
-        axis = (axis,)
-    items = 1
-    for ax in axis:
-        items *= arr.shape[mu.normalize_axis_index(ax, arr.ndim)]
+def _count_reduce_items(arr, axis, keepdims=False, where=True):
+    if isinstance(where, bool):
+        # no boolean mask given, calculate items according to axis
+        if axis is None:
+            axis = tuple(range(arr.ndim))
+        if not isinstance(axis, tuple):
+            axis = (axis,)
+        items = 1
+        for ax in axis:
+            items *= arr.shape[mu.normalize_axis_index(ax, arr.ndim)]
+    else:
+        # count True values in (potentially broadcasted) boolean mask
+        items = umr_sum(np.broadcast_to(where, arr.shape), axis, nt.intp, None,
+                        keepdims)
     return items
 
+
```

Comment:
This extra blank line was unintended. I will remove it with possible changes after the next review.